app/lib/admin/backend.dart

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:client_data/admin_api.dart' as api;
+import 'package:continuation_token/continuation_token.dart';
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+
+import '../account/backend.dart';
+import '../account/models.dart';
+import '../package/models.dart';
+import '../publisher/models.dart';
+import '../shared/configuration.dart';
+import '../shared/exceptions.dart';
+
+final _logger = Logger('pub.admin.backend');
+
+/// Sets the admin backend service.
+void registerAdminBackend(AdminBackend backend) =>
+    ss.register(#_adminBackend, backend);
+
+/// The active admin backend service.
+AdminBackend get adminBackend => ss.lookup(#_adminBackend) as AdminBackend;
+
+/// Represents the backend for the admin handling and authentication.
+class AdminBackend {
+  final DatastoreDB _db;
+  AdminBackend(this._db);
+
+  Future<R> _withAdmin<R>(Future<R> fn(AuthenticatedUser user)) async {
+    return await withAuthenticatedUser((u) async {
+      // TODO: remove once [withAuthenticatedUser] is using [User]
+      final user = await accountBackend.lookupUserById(u.userId);
+      if (!activeConfiguration.admins.contains(user.oauthUserId)) {
+        throw AuthorizationException.userIsNotAdminForPubSite();
+      }
+      return await fn(u);
+    });
+  }
+
+  /// List users.
+  Future<api.AdminListUsersResponse> listUsers(
+      {String continuationToken, int limit = 1000}) async {
+    return await _withAdmin((user) async {
+      final query = _db.query<User>()
+        ..order('__key__')
+        ..limit(limit);
+
```

Comment:
`InvalidInputException.checkRange(limit, 'limit', maximum: 1000, minimum: 1);` or something like that..